src/socket/address.cr

```diff
@@ -213,13 +252,23 @@ class Socket
     end
 
     def to_s(io)
+      if abstract?
+        io << '@'
+      end
       io << path
     end
 
     def to_unsafe : LibC::Sockaddr*
       sockaddr = Pointer(LibC::SockaddrUn).malloc
       sockaddr.value.sun_family = family
-      sockaddr.value.sun_path.to_unsafe.copy_from(@path.to_unsafe, @path.bytesize + 1)
+
+      destination = sockaddr.value.sun_path.to_unsafe
```

Comment:
`sun_path` is a `LibC::Char[?]`: Use a Slice with `#to_slice` instead of raw `#to_unsafe` Pointer.